feat: peer id smart contract lookup (#2)
CheckerNetwork/index-provider-peer-id: README.md

# index-provider-peer-id

Functionality for fetching index provider peer IDs for a minder id on the Filecoin network.

## Constants

### Smart Contract Integration

This library interacts with the MinerPeerIDMapping smart contract on the Filecoin EVM to retrieve
peer ID data.

#### `MINER_TO_PEERID_CONTRACT_ADDRESS`

The Ethereum address of the deployed MinerPeerIDMapping contract:

> `0x14183aD016Ddc83D638425D6328009aa390339Ce`

Reviewer: +1

#### Contract ABI

The minimal ABI containing just the `getPeerData` method used to fetch peer ID and signature data
for a given miner ID:

```solidity
function getPeerData(uint64 minerID) view returns (tuple(string peerID, bytes signature))

> ```

Reviewer: +1

> ## Peer ID Lookup

Reviewer 2: It is convention for the first heading to be the project's name

Reviewer 2: Also, the top heading should be a h1 / #

Functionality for fetching index provider peer IDs for a minder id on the Filecoin network.

### Provider Initialization

> Before using the functions fetching the peer IDs, you'll need to initialize an Filecoin provider and

Reviewer 2: ```suggestion
Before using the functions fetching the peer IDs, you'll need to initialize a Filecoin JSON-RPC provider and
```

> contract instance. Here is an example using a `JsonRpcProvider` from the `ethers.js` library. You
> can use other providers as well, as long as they are compatible with the `ethers.js` library.

Reviewer 2: ```suggestion
contract instance. Here is an example using a `JsonRpcProvider`:
```

Which ethers.js provider to use is not the responsibility of this library

Author: What do you mean? The idea was to show an example of how to use the library's content. In this case the ABI and contract address.

```javascript
import { ethers } from 'ethers'
import { ABI, MINER_TO_PEERID_CONTRACT_ADDRESS } from 'index-provider-peer-id'

> // Example rpc provider url for Glif node
> const RPC_URL = 'https://api.node.glif.io/'
> 
> // Optional: authentication token for the RPC provider
> const RPC_AUTH = '<YOUR_TOKEN>'
> 
> // Create a custom JsonRpcProvider with authorization header if auth token is provided
> const fetchRequest = new ethers.FetchRequest(RPC_URL)
> 
> if (RPC_AUTH) {
>   fetchRequest.setHeader('Authorization', `Bearer ${RPC_AUTH}`)
> }
> 
> const provider = new ethers.JsonRpcProvider(fetchRequest)

Reviewer 2: ```suggestion
// Example rpc provider url for Glif node
const provider = new ethers.JsonRpcProvider('https://api.node.glif.io/')
```

Configuring an ethers provider is out of scope for this library

Author: As a user I would appreciate if I just had to use the code examples that are given for a particular library instead of having to look for other sources again. Why do you think this is out of scope?

> return new ethers.Contract(MINER_TO_PEERID_CONTRACT_ADDRESS, ABI, provider)

Reviewer 2: why is this returned / what is it being returned to?
